commons/all/src/main/java/org/infinispan/commons/client/hotrod/metric/HotRodMetricRegistry.java

```diff
@@ -0,0 +1,79 @@
+package org.infinispan.commons.client.hotrod.metric;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import org.infinispan.commons.stat.CounterTracker;
+import org.infinispan.commons.stat.DistributionSummaryTracker;
+import org.infinispan.commons.stat.TimerTracker;
+
+/**
+ * Hot Rod client entrypoint.
+ * <p>
+ * All metrics registered in this instance are considered global to the client. Per cache metrics are registered using
+ * {@link #withCache(String)}, where the cache's name is used to distinct the metrics. The implementation is responsible
+ * to separate each cache metric, because all cache register the same set of metrics (reads, writes, etc.).
+ *
+ * @since 15.1
+ */
+public interface HotRodMetricRegistry extends MetricRegistry {
```

Comment:
Why are the HR specific classes in commons?

Reply:
To be shared between the client implementation. The alternative is to create a new module.

Reply:
Ok, I was forgetting that we had the new implementation arriving :+1: